File: backend/src/main/java/com/buysell/modules/device/entity/Device.java
package com.buysell.modules.device.entity;

import com.buysell.modules.device.enums.DeviceStatus;
import com.buysell.modules.user.entity.User;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.SuperBuilder;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.OffsetDateTime;
import java.util.UUID;

@Entity
@Table(name = "devices")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
public class Device {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "imei1", nullable = false, unique = true, length = 50)
    private String imei1;

    @Column(name = "imei2", length = 50)
    private String imei2;

    @Column(name = "serial_number", length = 100)
    private String serialNumber;

    @Column(name = "brand", nullable = false, length = 100)
    private String brand;

    @Column(name = "model", nullable = false, length = 100)
    private String model;

    @Column(name = "variant", length = 100)
    private String variant;

    @Column(name = "color", length = 100)
    private String color;

    @Column(name = "storage_gb")
    private Integer storageGb;

    @Column(name = "ram_gb")
    private Integer ramGb;

    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false, length = 20)
    @Builder.Default
    private DeviceStatus status = DeviceStatus.ACTIVE;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by")
    private User createdBy;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "updated_by")
    private User updatedBy;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private OffsetDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private OffsetDateTime updatedAt;
}

File: android/app/src/main/java/com/mobile/app/domain/model/device/Device.kt
package com.mobile.app.domain.model.device

import java.time.LocalDateTime

data class Device(
    val id: String,
    val brand: String,
    val model: String,
    val variant: String?,
    val color: String,
    val storage: String,
    val ram: String,
    val imei1: String,
    val imei2: String?,
    val serialNumber: String?,
    val status: DeviceStatus,
    val branchId: String,
    val createdBy: String,
    val updatedBy: String?,
    val createdAt: LocalDateTime,
    val updatedAt: LocalDateTime,
    val verificationState: ImeiVerificationState,
    val mediaCount: Int,
    val latestCondition: DeviceCondition?,
    val latestInspection: DeviceInspection?
)

data class DeviceCreate(
    val brand: String,
    val model: String,
    val variant: String?,
    val color: String,
    val storage: String,
    val ram: String,
    val imei1: String,
    val imei2: String?,
    val serialNumber: String?
)

data class DeviceUpdate(
    val brand: String,
    val model: String,
    val variant: String?,
    val color: String,
    val storage: String,
    val ram: String,
    val imei1: String,
    val imei2: String?,
    val serialNumber: String?
)
